.github/workflows/mlflow-integration-tests.yml

```diff
@@ -0,0 +1,57 @@
+name: MLflow-UC OSS Integration Test
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -exo pipefail {0}
+
+env:
+  MLFLOW_HOME: /home/runner/work/mlflow/mlflow
+  UC_OSS_INTEGRATION: true
+
+jobs:
+  uc-oss-integration-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.9', '3.10']
+    timeout-minutes: 120
+    permissions: {}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          pip install mlflow
+          pip install pytest
+      - name: Start Unity Catalog Server
+        run: |
+          ai/dev/start_unitycatalog.sh
+      - name: Run tests for UnityCatalog
+        run: |
+          echo "Running tests for UnityCatalog..."
+          pytest tests/test_mlflow_integration.py
+      - name: Stop Unity Catalog Server
+        if: always()
```

Comment:
Do we want to add this? IIRC, this makes the step/run not cancellable.

Reply:
We can. I do see that github actually will just kill orphaned processes after the workflow ends though, so while good practice to clean up, I don't think there is a risk of tons of orphaned servers sticking around.